Author: illusionist
E Mail: arch(dawt)illusionist(at)gmail(dawt)com
Patch Description:
This patch send programs to last tag (defined as background tag, this tag wont be visible on the status bar.)
Toggle applications with the defined key. 
Relevent parts of config.h:
===========================
static const Rule rules[] = {
	/* class      		instance	title		tags mask	isfloating 	iscenterd	monitor */
	{ NULL,			NULL,		"wget",		1 << 5,		True,		True,  		-1 },
};
static Key keys[] = {
	{ Modkey,                       	XK_space,      		toggleview,     	{.ui = 1 << 5} },	
	{ Modkey|Shiftkey,					XK_space,			tag,				{.ui = 1 << 5} },
}
--- dwm.c.orig	2012-12-12 09:16:07.044687928 +0530
+++ dwm.c	2013-01-12 14:28:08.248429526 +0530
@@ -514,8 +514,8 @@ buttonpress(XEvent *e) {
 		i = x = 0;
 		do
 			x += TEXTW(tags[i]);
-		while(ev->x >= x && ++i < LENGTH(tags));
-		if(i < LENGTH(tags)) {
+		while(ev->x >= x && ++i < (LENGTH(tags) - 1));
+		if(i < (LENGTH(tags) - 1)) {
 			click = ClkTagBar;
 			arg.ui = 1 << i;
 		}
@@ -868,7 +868,7 @@ drawbar(Monitor *m) {
 			urg |= c->tags;
 	}
 	dc.x = 0;
-	for(i = 0; i < LENGTH(tags); i++) {
+	for(i = 0; i < (LENGTH(tags) - 1); i++) {
 		dc.w = TEXTW(tags[i]);
 		col = dc.colors[ (m->tagset[m->seltags] & 1 << i) ? 1 : (urg & 1 << i ? 2:(occ & 1 << i ? 3:0)) ];
         drawtext(tags[i], col, True);
@@ -2658,7 +2658,7 @@ reload(const Arg *arg) {
 void
 cycle(const Arg *arg) {
     int seltags = selmon->tagset[selmon->seltags];
-    int numtags = LENGTH(tags);
+    int numtags = (LENGTH(tags) - 1);
     int delta = arg->i;
     int curtag = -1;
 
@@ -2673,7 +2673,7 @@ cycle(const Arg *arg) {
 void
 tagcycle(const Arg *arg) {
     int seltags = selmon->tagset[selmon->seltags];
-    int numtags = LENGTH(tags);
+    int numtags = (LENGTH(tags) - 1);
     int delta = arg->i;
     int curtag = -1;
 
